firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -271,6 +275,14 @@ String getName() {
     return Tasks.call(backgroundExecutor, this::deleteFirebaseInstallationId);
   }
 
+  /** Register a callback {@link FidListener} to receive fid changes. */
+  @Override
+  public void registerFidListener(@Nullable FidListener listener) {
+    synchronized (this) {
+      this.fidListener = listener;
```

Comment:
Looks like you're only supporting one listener at a time, I wonder if we want to allow multiple listeners, otherwise it may be surprising to clients that their listener is removed just because someone else registered a listener as well.